ehr_ml/clmbr/dataset.py

```diff
@@ -28,24 +43,24 @@ def prepare_batch_thread(
             out_queue.put(None)
             break
 
-        result = transform_func(item)
+        result = finalize_data(item, device)
         out_queue.put(result)
 
-
-class BatchIterator:
+class DataLoader:
     def __init__(
         self,
         dataset: StrideDataset,
-        transform_func: Any,
         threshold: int,
         is_val: bool = False,
         batch_size: int = 2000,
         seed: int = 0,
         day_dropout: float = 0,
         code_dropout: float = 0,
+        use_cuda: bool = torch.cuda.is_available()
```

Comment:
It would probably be better to have this do use_code = None and then call cuda.is_available() inside the function. Calling cuda functions during startup is suboptimal.